libs/queue/SPECIFICATION.md

```diff
@@ -4,32 +4,34 @@ This document provides an overview of the Queue library.
 
 It outlines the use cases, i.e. specification, and describes how to implement the library.
 
+The `Queue<T>` is a linear structure with a generic type parameter `T`. The elements in the queue are stored as a `Vec<T>`. Queues are growable, operate in a First-In-First-Out (FIFO) manner, and are created empty.
+
 ## Use Cases
 
-The Queue library can be used as a traditional queue or first in first out data structure.
+The Queue library can be used First-In-First-Out data structure when you want to get items out in the order that they are added.
```

Comment:
This is basically repeating the first line of the readme.

You could probably expand on / simplify the part about retrieving items in the order that they are added.